MAINT - Core and apps compiler warnings

carlwilson opened the pull request:
- removed unthrown exceptions in method definitions;
- removed unused variables, code, and imports, including `PropertyPath.java`;
- made methods static where trivial;
- replaced some `for (int i...)` loops with `for (x : y)` style;
- added try with resources for file and stream handling where trivial;
- explicit types for generic declarations;
- fixed `variable.equals(constant)` to be `constant.equals(variable)` to protect against null pointer errors;
- removed associated and unnecessary casts; and
- added missing `@Override` annotations.

Addresses #370

carlwilson committed:
MAINT - Core and apps compiler warnings
- removed unthrown exceptions in method definitions;
- removed unused variables, code, and imports, including `PropertyPath.java`;
- made methods static where trivial;
- replaced some `for (int i...)` loops with `for (x : y)` style;
- added try with resources for file and stream handling where trivial;
- explicit types for generic declarations;
- fixed `variable.equals(constant)` to be `constant.equals(variable)` to protect against null pointer errors;
- removed associated and unnecessary casts; and
- added missing `@Override` annotations.

carlwilson committed:
FIX - General Exception for Module resetParams
- replaced general exception throw on `resetParams` while compatability with old modules is tested.

carlwilson committed:
Merge branch 'integration' into refact/core-apps-warnings

david-russo commented on jhove-apps/src/main/java/edu/harvard/hul/ois/jhove/viewer/InfoWindow.java:
```diff
@@ -131,14 +132,14 @@ protected PrintWriter doSaveDialog ()
-        Iterator iter = handlerList.iterator ();
+        Vector<String> handlerItems = new Vector<> (10);
+        java.util.List<?> handlerList = _base.getHandlerList ();
+        Iterator<?> iter = handlerList.iterator ();
```
For the current code, this List seems intended to only ever hold OutputHandlers. Is the wildcard use preparation for a future change? If so, might still want to bundle that together with those future changes.

And since we're changing the Iterator definition, let's take it a step further and change it all the way out of existence :)
```
List<OutputHandler> handlerList = _base.getHandlerList();
for (OutputHandler han : handlerList) {
    ...
```

david-russo commented on jhove-apps/src/main/java/edu/harvard/hul/ois/jhove/viewer/RepTreeRoot.java:
```diff
@@ -649,7 +644,7 @@ private DefaultMutableTreeNode aesToNode (AESAudioMetadata aes)
         // In the general case, it can contain multiple Faces;
         // this isn't supported yet.
-        List facelist = aes.getFaceList ();
+        List<?> facelist = aes.getFaceList ();
```
`List<AESAudioMetadata.Face>`

david-russo commented on jhove-apps/src/main/java/edu/harvard/hul/ois/jhove/viewer/RepTreeRoot.java:
```diff
@@ -693,7 +688,7 @@ private DefaultMutableTreeNode aesToNode (AESAudioMetadata aes)
         // modules; if it's needed in the future, simply set up an
         // iteration loop on formatList.
-        List flist = aes.getFormatList ();
+        List<?> flist = aes.getFormatList ();
```
`List<AESAudioMetadata.FormatRegion>`

david-russo commented on jhove-core/src/main/java/edu/harvard/hul/ois/jhove/handler/AuditHandler.java:
```diff
@@ -232,7 +224,7 @@ public void showImpl (RepInfo info, AuditState state)
 
 	String md5 = null;
-	List list = info.getChecksum ();
+	List<?> list = info.getChecksum ();
```
`List<Checksum>`

david-russo reviewed (changes requested):
Good stuff, just a few changes to keep the type checker working for us where it can.

carlwilson committed:
REV - Addressed review comments
- adressed missed generic type declarations;
- accepted loop suggestion; and
- picked up some other loops.

carlwilson committed:
Merge branch 'integration' into refact/core-apps-warnings
